src/extension/src/azure/azure-resource-group/resourceGroupModule.ts

```diff
@@ -47,29 +48,42 @@ export class ResourceGroupDeploy {
         location: location
       };
 
-      const result = await this.azureResourceClient.resourceGroups.createOrUpdate(
+      await this.azureResourceClient.resourceGroups.createOrUpdate(
         resourceGroupName,
         options
       );
-      return result.name;
     } catch (error) {
       throw new DeploymentError(error.message);
     }
   }
 
   public async generateValidResourceGroupName(
     name: string,
-    userSubscriptionItem: SubscriptionItem
+    userSubscriptionItems: SubscriptionItem[]
   ): Promise<string> {
-    this.setAzureResourceClient(userSubscriptionItem);
-
     let generatedName = this.generateResourceGroupName(name);
-    let isValid = await this.validateResourceGroupName(generatedName);
+    let isValid: boolean = true;
+    // this allows the generated name to be validated against multiple subscriptions
+    userSubscriptionItems.forEach(async userSubscriptionItem => {
```

Comment:
nit why repeat this code here and in lines 79-86? The lines look very similar if not the same.

Reply:
because I need to know if the first generated name is valid, if it is not, then the function will re-generate a name and re-validate the new name within the loop

Reply:
Why not just go into the loop right away by setting isValid to false?